r2i/tensorflow/frame.cc

```diff
@@ -121,12 +121,14 @@ RuntimeError Frame::Validate (int64_t dims[], int64_t num_dims) {
     return error;
   }
 
-  /* Check that channels match
-   * TODO: relate this to the input format
-   */
-  if (3 != dims[3]) {
-    error.Set (RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
-               "We only support a 3 channels per image");
+  /* Check that channels match */
+  int frame_format_channels = this->frame_format.GetNumPlanes();
+  if (frame_format_channels != dims[3]) {
+    char msg[70];
+    sprintf(msg,
```

Comment:
Try to avoid `sprintf`. Prefer `snprintf`. Or, since we are in C++`, use std::string.